.gitignore

```diff
@@ -1,7 +1,33 @@
-/.idea
+# TODO: start including composer.lock
 composer.lock
-vendor/
-docs/
-.editorconfig
+
+# OSX
+**/.DS_Store
+
+# IDEs
+.project/
+.settings/
+.idea/
+*.code-workspace
+*.sublime-project
+*.sublime-workspace
+*.iml
+.vscode/
```

Comment:
I would argue that the repo should not contain development environment specific ignores. Contributors can/should set their own local global ignores.

Reply:
I think you're definitely technically correct on that one. But with an open source repo that may have any number of contributors (including those that haven't configured their global ignore), maybe this just makes things easier?

Reply:
Yes my thinking was the same as @agibson-godaddy